src/Resources/skeleton/controller/ControllerWithTwig.php.txt

```diff
@@ -13,7 +13,6 @@ class {{ controller_class_name }} extends Controller
      */
     public function index()
     {
-        // replace this line with your own code!
-        return $this->render('@Maker/demoPage.html.twig', [ 'path' => str_replace($this->getParameter('kernel.project_dir').'/', '', __FILE__) ]);
+        return $this->render('your_own_template.html.twig');
```

Comment:
actually, it would be great to add a second argument to hint you can pass vars to it,
and also, would be greate to generate the twig file, using a simple naming scheme (similar to the one used by the `@Template` annotation?.)
The content of the file could be trivial, eg:
```
{% extends base.html.twig %}
{% block body %}
Very simple welcome message, or next steps hint
{% block endblock %}
```

Reply:
We "can't" do that because maybe this controller is used in a API app that returns JSON contents. We can't assume Twig, templates, etc.  I know that the `$this->render()` call assumes that ... but it's simple to remove, unlike having to look for and remove a separate Twig file, etc.

Reply:
It's trivial to remove, and as you said is already opinionated since "render" is called.
The currently generated code it broken, which is bad experience.
Just dropping an additional file and the experience becomes smooth.

Reply:
I agree with @nicolas-grekas here. Generating some code that triggers an error is not a good DX.

Reply:
the example given by @nicolas-grekas still assumes a `base.html.twig` template, which may not be there either.

And btw, if Twig is not installed, this would also stay broken.

Reply:
I agree with @nicolas-grekas and @stof here, should be a simple template without `extends` tag.

> And btw, if Twig is not installed, this would also stay broken.

this template is only loaded if Twig dep is available.

Reply:
*with* the extends tag, so that discovering the base template is made easier, and valid html pages are generated by default :)

Reply:
why not just `return new Reponse('...')`?

Reply:
Because this is the controller that is generated when twig is installed.

Reply:
I don't see why we should distinguish that in the first place. Just because twig is installed, you still don't know what the controller is meant for. So just do the MVP.

Reply:
We're doing all of this to help the 80% use case, and to help newcomers discover how this is easy to use (and discover)
I think it's not unreasonable (like >50% OK) to think that if you have twig, you might want to use it to create your pages. That's the reasoning, and I fully support it :)